hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -418,13 +418,16 @@ boolean needsBalance(TableName tableName, BalancerClusterState cluster) {
     }
 
     double total = 0.0;
+    float localSumMultiplier = 0; // in case sumMultiplier is not initialized
     for (CostFunction c : costFunctions) {
       if (!c.isNeeded()) {
         LOG.trace("{} not needed", c.getClass().getSimpleName());
         continue;
       }
       total += c.cost() * c.getMultiplier();
+      localSumMultiplier += c.getMultiplier();
     }
+    sumMultiplier = localSumMultiplier;
```

Comment:
This is a small change, but sumMultiplier is initialized in `balanceTable`, and `needsBalance` is both visible to the package, and could reasonably be called _before_ balancing. We're iterating through cost functions anyway here, we might as well have a local copy of the total multiplier